crates/cranelift/src/obj.rs

```diff
@@ -331,19 +345,42 @@ impl<'a> UnwindInfoBuilder<'a> {
     /// anything on the other end of loading a module from a precompiled object.
     ///
     /// Support for reading this is in `crates/jit/src/unwind/winx64.rs`.
-    fn write_windows_unwind_info(&self, obj: &mut Object<'_>, section_id: SectionId) {
+    fn write_windows_unwind_info(
+        &self,
+        obj: &mut Object<'_>,
+        xdata_id: SectionId,
+        pdata_id: SectionId,
+        text_section_size: u64,
+    ) {
         // Currently the binary format supported here only supports
         // little-endian for x86_64, or at least that's all where it's tested.
         // This may need updates for other platforms.
         assert_eq!(obj.architecture(), Architecture::X86_64);
 
-        let mut unwind_info = Vec::with_capacity(self.windows_unwind_info.len() * 3 * 4);
-        for info in self.windows_unwind_info.iter() {
-            unwind_info.extend_from_slice(&info.begin.to_le_bytes());
-            unwind_info.extend_from_slice(&info.end.to_le_bytes());
-            unwind_info.extend_from_slice(&info.unwind_address.to_le_bytes());
+        // Append the `.xdata` section, or the actual unwinding information
+        // codes and such which were built as we found unwind information for
+        // functions.
+        obj.append_section_data(xdata_id, &self.windows_xdata, 4);
+
+        // Next append the `.pdata` section, or the array of `RUNTIME_FUNCTION`
+        // structures stored in the binary.
+        //
+        // Note that this section has a "relocation" applied here per-entry
```

Comment:
It's fairer to say that every address in the `RUNTIME_FUNCTION` structure must be start-of-text-section relative (thus necessitating that `.xdata` comes _after_ the text section) because we're using `RtlAddFunctionTable` to register the `.pdata` section contents and using the loaded text section's address as the "image base address".

I think mentioning "relocations" here might be confusing since technically no fixups are being recorded in the artifact to apply at load time (unlike what a real `.pdata` section would have in a PE since `RtlAddFunctionTable` wouldn't be used in that case).